lib/auth0/appEndpointHandler.js

```diff
@@ -6,11 +6,15 @@ module.exports = function (webtask, options, ctx, req, res, routingInfo) {
     function authenticate() {
         var token = options.getAccessToken(ctx, req);
         if (!token) {
+            var stateInfo = {
+                appPath: routingInfo.appPath,
+                query: routingInfo.query
+            };
             return options.loginError({
                 code: 401,
                 message: 'Unauthorized.',
                 error: 'Missing access token.',
-                redirect: routingInfo.baseUrl + '/login'
+                redirect: routingInfo.baseUrl + '/login?state=' + encodeURIComponent(JSON.stringify(stateInfo))
```

Comment:
Ahhh, I think you're right. 

The security issue is a good point, and using a JWT is a great suggestion. Thanks for the link to the IETF draft.

So, probably a better approach that meets the opt-in desire would be to add a 'state' element to the `options` object. This would be a function that returns an opaque state object that gets wrapped up into a JWT by the **webtask-tools** package.

The **webtask-tools** package would also decode the JWT on the other side, and make the state object available via the `loginSuccess()` interface upon completion of a login sequence.

The default implementation of `options.state` could be to simply return a null value, in which case no state information would be carried around, mimicking the current behavior.

Something like that, yes? It seems closer to the design patterns already in place...